jmix-flowui/flowui-data/src/main/java/io/jmix/flowuidata/genericfilter/FlowuiDataGenericFilterSupport.java

```diff
@@ -0,0 +1,292 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.flowuidata.genericfilter;
+
+import com.google.common.collect.ImmutableSet;
+import io.jmix.core.DataManager;
+import io.jmix.core.Metadata;
+import io.jmix.core.annotation.Internal;
+import io.jmix.core.querycondition.LogicalCondition;
+import io.jmix.core.querycondition.PropertyCondition;
+import io.jmix.core.security.CurrentAuthentication;
+import io.jmix.core.security.event.UserRemovedEvent;
+import io.jmix.flowui.Actions;
+import io.jmix.flowui.UiComponents;
+import io.jmix.flowui.action.genericfilter.GenericFilterAction;
+import io.jmix.flowui.action.genericfilter.GenericFilterClearValuesAction;
+import io.jmix.flowui.action.genericfilter.GenericFilterCopyAction;
+import io.jmix.flowui.action.genericfilter.GenericFilterEditAction;
+import io.jmix.flowui.app.filter.condition.FilterConditionDetailView;
+import io.jmix.flowui.component.genericfilter.Configuration;
+import io.jmix.flowui.component.genericfilter.GenericFilter;
+import io.jmix.flowui.component.genericfilter.GenericFilterSupport;
+import io.jmix.flowui.component.genericfilter.configuration.AbstractConfigurationDetail;
+import io.jmix.flowui.component.logicalfilter.LogicalFilterComponent;
+import io.jmix.flowui.model.DataComponents;
+import io.jmix.flowui.model.InstanceContainer;
+import io.jmix.flowui.model.ViewData;
+import io.jmix.flowui.view.DialogWindow;
+import io.jmix.flowui.view.StandardOutcome;
+import io.jmix.flowui.view.ViewControllerUtils;
+import io.jmix.flowuidata.action.genericfilter.GenericFilterRemoveAction;
+import io.jmix.flowuidata.action.genericfilter.GenericFilterSaveAction;
+import io.jmix.flowuidata.action.genericfilter.GenericFilterSaveAsAction;
+import io.jmix.flowuidata.action.genericfilter.GenericFilterSaveWithValuesAction;
+import io.jmix.flowuidata.component.genericfilter.configuration.FlowuiDataFilterConfigurationDetail;
+import io.jmix.flowuidata.entity.FilterConfiguration;
+import jakarta.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.transaction.event.TransactionPhase;
+import org.springframework.transaction.event.TransactionalEventListener;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import static io.jmix.flowui.component.genericfilter.FilterUtils.generateFilterPath;
+
+@Internal
+public class FlowuiDataGenericFilterSupport extends GenericFilterSupport {
```

Comment:
Something odd with Configuration editor. Checkboxes do not handles their label themself, as a result they are misaligned.

<img width="841" alt="Screenshot 2023-05-29 at 17 23 01" src="https://github.com/jmix-framework/jmix/assets/6077149/0562deb3-0c68-4ffd-9f13-6dfd92167ed7">

See a default defauilt view as a reference:

<img width="1200" alt="Screenshot 2023-05-29 at 17 23 09" src="https://github.com/jmix-framework/jmix/assets/6077149/6c2603d5-41af-4be5-8ddd-195f3cf05cd5">